src/setup_jest.ts

```diff
@@ -26,3 +28,32 @@ if (typeof window !== "undefined") {
 
   window.open = jest.fn()
 }
+
+/**
+ * Fail tests that log errors or warnings, because these can point to actual
+ * bugs and once there are already a few of these the person writing new code
+ * will start ignoring them.
+ *
+ * If a test is expected to log an error or a warning, mock it so the output
+ * doesn’t actually show up.
+ */
+const logAndThrow = loggerFn => {
+  // tslint:disable-next-line:only-arrow-functions
+  const imp = function(message) {
+    // Keep default logging behaviour
+    loggerFn.apply(console, arguments)
+    if (message instanceof Error) {
+      throw message
+    } else {
+      const err = new Error(message)
+      // Skip this frame in the stack to point to the actual log call-site
+      Error.captureStackTrace(err, imp)
+      throw err
+    }
+  }
+  return imp
+}
+const originalConsoleError = console.error
+const originalWarnError = console.warn
+console.error = logAndThrow(originalConsoleError)
+console.warn = logAndThrow(originalWarnError)
```

Comment:
this is a great technique, I kinda want it in more places

Reply:
👏 YES! I love this! 👏

Reply:
🤫🌈🌟🤩